impl/src/test/java/io/quarkiverse/kafkastreamsprocessor/impl/TracingQuarkusTest.java

```diff
@@ -0,0 +1,204 @@
+                new KafkaProtobufSerializer<>());
+        Map<String, Object> consumerProps = KafkaTestUtils.consumerProps(bootstrapServers, "test", "true");
+        consumer = new KafkaConsumer<>(consumerProps, new StringDeserializer(),
+                new KafkaProtobufDeserializer<>(PingMessage.Ping.parser()));
+        consumer.subscribe(List.of(consumerTopic));
+
+        clearSpans();
+    }
+
+    private void clearSpans() {
+        // force a flush to make sure there are no remaining spans still in the buffers
+        ((OpenTelemetrySdk) openTelemetry).getSdkTracerProvider().forceFlush();
+        testSpanExporter.getSpans().clear();
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        clearSpans();
+        producer.close();
+        consumer.close();
+    }
+
+    @Test
+    public void tracingShouldBePropagatedW3C() {
+        Span parentSpan = tracer.spanBuilder("parent").startSpan();
+        try (Scope parentScope = parentSpan.makeCurrent()) {
+            PingMessage.Ping input = newBuilder().setMessage("world").build();
+            RecordHeaders headers = new RecordHeaders();
+            openTelemetry.getPropagators()
+                    .getTextMapPropagator()
+                    .inject(Context.current(), headers, kafkaTextMapSetter);
+            ProducerRecord<String, PingMessage.Ping> record = new ProducerRecord<>(senderTopic, 0, "key", input, headers);
+            producer.send(record);
+
+            ConsumerRecord<String, PingMessage.Ping> singleRecord = KafkaTestUtils.getSingleRecord(consumer, consumerTopic,
+                    Duration.ofSeconds(5));
+            MatcherAssert.assertThat(toMap(singleRecord.headers()),
+                    hasEntry(equalTo(KafkaStreamsProcessorHeaders.W3C_TRACE_ID),
+                            containsString(parentSpan.getSpanContext().getTraceId())));
+        } finally {
+            parentSpan.end();
+        }
+    }
+
+    private static Map<String, String> toMap(Iterable<Header> headers) {
+        Map<String, String> result = new HashMap<>();
+        for (Header h : headers) {
+            result.put(h.key(), new String(h.value(), StandardCharsets.UTF_8));
+        }
+        return result;
+    }
+
+    @Test
+    public void spanShouldBeCreatedW3C() {
+
+        Span parentSpan = tracer.spanBuilder("parent").startSpan();
+        try (Scope ignored = parentSpan.makeCurrent()) {
+            PingMessage.Ping input = newBuilder().setMessage("world").build();
+            RecordHeaders headers = new RecordHeaders();
+            openTelemetry.getPropagators()
+                    .getTextMapPropagator()
+                    .inject(Context.current(), headers, kafkaTextMapSetter);
+            ProducerRecord<String, PingMessage.Ping> record = new ProducerRecord<>(senderTopic, 0, "key", input, headers);
+            producer.send(record);
+
+            KafkaTestUtils.getSingleRecord(consumer, consumerTopic, Duration.ofSeconds(5000));
```

Comment:
Typo in the timeout value here I suppose?

Also, I don't see the difference between the two tests. Can't you merge them and assert on both creation and propagation at once?